### PR TITLE
Kotlin Synthetics removal

### PR DESCRIPTION
**Pull request by avazirna**
## Summary
[Kotlin Android Extensions Gradle plugin](https://plugins.gradle.org/plugin/org.jetbrains.kotlin.android.extensions) was deprecated in Nov 2020 and the removal set to Kotlin 1.8, which was released in Dec 2022. This ticket is to migrate Kotlin Synthetics to the new recommended approach, View Binding.

## Safety Assurance
- [X] If the PR is high risk, "High Risk" label is set
- [X] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story
- This PR only affects features using MapBox map, which are being used by a few projects. 
- A small note for reviewers related to the removal of the UI Controller, considering that View Binding offers a more elegant and safer way to access views, it seemed partially redundant to keep the controller, however, I wonder if there are other aspects that I might be neglecting here.

### CODE REVIEW

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

File: app/src/org/commcare/gis/DrawingBoundaryActivity.kt [hunk 1]
```diff
@@ -250,4 +257,54 @@ class DrawingBoundaryActivity : BaseMapboxActivity(), WithUIController, Location
...
+    private fun initUI() {
+        viewBinding.startTrackingButton.text =  Localization.get("drawing.boundary.map.start.tracking")
+        viewBinding.startTrackingButton.setOnClickListener {
+            this.startTracking()
+        }
+
+        viewBinding.stopTrackingButton.text =  Localization.get("drawing.boundary.map.stop.tracking")
+        viewBinding.stopTrackingButton.setOnClickListener {
+            stoppedUIState()
+            this.stopTracking()
```
Comment (shubham1g5) [inline app/src/org/commcare/gis/DrawingBoundaryActivity.kt:278]: nit: replace `this.x` -> `x` in this method

File: app/src/org/commcare/gis/EntityMapboxActivity.kt [hunk 2]
```diff
@@ -81,10 +84,17 @@ class EntityMapboxActivity : BaseMapboxActivity() {
...
+    override fun getViewBinding(): ViewBinding {
+        if (!::viewBinding.isInitialized) {
+            viewBinding = ActivityEntityMapboxBinding.inflate(layoutInflater)
+        }
+        return viewBinding
+    }
 
-    override fun getMapLayout(): Int {
-        return R.layout.activity_entity_mapbox
+    override fun getMapView(): KujakuMapView {
```
Comment (shubham1g5) [inline app/src/org/commcare/gis/EntityMapboxActivity.kt:96]: is this same as it's super method in `BaseMapboxActivity` ?

**Review by shubham1g5 — APPROVED** [review]
Looks quite solid!

File: app/src/org/commcare/gis/EntityMapboxActivity.kt (hunk 2 above)
Comment (avazirna) [reply app/src/org/commcare/gis/EntityMapboxActivity.kt:96]: Yes. Also, this method is abstract in `BaseMapboxActivity`

File: app/src/org/commcare/gis/DrawingBoundaryActivity.kt [hunk 3]
```diff
@@ -115,20 +130,20 @@ class DrawingBoundaryActivity : BaseMapboxActivity(), WithUIController, Location
     }
 
     private fun onStyleLoaded() {
-        mapView.isWarmGps = true
-        drawingManager = DrawingManager(mapView, map, loadedStyle)
+        viewBinding.mapView.isWarmGps = true
```
[the remark this answers is not in the document]
Comment (shubham1g5) [reply app/src/org/commcare/gis/DrawingBoundaryActivity.kt]: there are a bunch of other places in this PR where we are using `viewBinding.mapView`, we should either replace them with `getMapView()` or remove `getMapView()` and use `viewBinding.mapView` everywhere.

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**